src/utilities/with-app-provider.tsx

```diff
@@ -52,66 +53,32 @@ export function withAppProvider<OwnProps>({withinScrollable}: Options = {}) {
   return function addProvider<C>(
     WrappedComponent: ReactComponent<OwnProps & WithAppProviderProps, C>,
   ): React.ComponentClass<OwnProps> & C {
-    // eslint-disable-next-line react/prefer-stateless-function
-    class WithProvider extends React.Component<OwnProps, never> {
-      static contextTypes = WrappedComponent.contextTypes;
+    const WithProvider: React.FunctionComponent = (props: OwnProps) => {
```

Comment:
Will this be inferred or do we need to explicitly set it?

Reply:
`React.FunctionComponent` adds a [few more things](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L503) over a regular function. Notably adding children to the props, and adding some of the static things you can hang off of a component like `WithProvider.contextTypes. In 99% of the time this distinction doesn't matter as we don't use any of those properties but I'm being paranoid here.

Extending from React.Component did [the same sort of thing](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L519)